client/tx/tx.go

```diff
@@ -335,3 +346,77 @@ type GasEstimateResponse struct {
 func (gr GasEstimateResponse) String() string {
 	return fmt.Sprintf("gas estimate: %d", gr.GasEstimate)
 }
+
+// makeAuxSignerData generates an AuxSignerData from the client inputs.
+func makeAuxSignerData(clientCtx client.Context, f Factory, msgs ...sdk.Msg) (tx.AuxSignerData, error) {
+	if f.SignMode() == signing.SignMode_SIGN_MODE_DIRECT {
+		return tx.AuxSignerData{}, sdkerrors.Wrap(errors.New("can't sign a aux tx in DIRECT mode"), "tipper")
+	}
+
+	b := NewAuxTxBuilder()
+	fromAddress, name, _, err := client.GetFromFields(clientCtx.Keyring, clientCtx.From, false)
+
+	if err != nil {
+		return tx.AuxSignerData{}, err
+	}
+
+	b.SetAddress(fromAddress.String())
+	if clientCtx.Offline {
+		b.SetAccountNumber(f.accountNumber)
+		b.SetSequence(f.sequence)
+	} else {
+		accNum, seq, err := clientCtx.AccountRetriever.GetAccountNumberSequence(clientCtx, fromAddress)
+		if err != nil {
+			return tx.AuxSignerData{}, err
+		}
+		b.SetAccountNumber(accNum)
+		b.SetSequence(seq)
+	}
+
+	err = b.SetMsgs(msgs...)
+	if err != nil {
+		return tx.AuxSignerData{}, err
+	}
+
+	if f.tip != nil {
+		if f.tip.Tipper == "" {
+			return tx.AuxSignerData{}, sdkerrors.Wrap(errors.New("tipper flag required"), "tipper")
+		} else {
```

Comment:
Maybe let's add this check in `func (f Factory) WithTips()`?

Reply:
present codebase is not throwing error inside the `NewFactoryCLI` func